### ISSUE TITLE
EksPodOperator doesn't work with deferrable=True

### ISSUE THREAD

**Issue by 0x26res** [apache/airflow#38280]
### Apache Airflow Provider(s)

amazon

### Versions of Apache Airflow Providers

apache-airflow-providers-amazon==8.7.1

### Apache Airflow version

2.7.2

### Operating System

ubuntu (or whatever MWAA provides)

### Deployment

Official Apache Airflow Helm Chart

### Deployment details

MWAA 2.7.2

### What happened

The EksPodOperator can submit a job, but the trigger fails later.

I think it's because the kube config is saved in a temp file that is gone by the time the operator tries to poll the task progress.

```
  File "/usr/local/airflow/.local/lib/python3.11/site-packages/airflow/providers/cncf/kubernetes/triggers/pod.py", line 145, in run
    pod = await hook.get_pod(
          ^^^^^^^^^^^^^^^^^^^
  File "/usr/local/airflow/.local/lib/python3.11/site-packages/airflow/providers/cncf/kubernetes/hooks/kubernetes.py", line 563, in get_pod
    async with self.get_conn() as connection:
  File "/usr/local/lib/python3.11/contextlib.py", line 204, in __aenter__
    return await anext(self.gen)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/airflow/.local/lib/python3.11/site-packages/airflow/providers/cncf/kubernetes/hooks/kubernetes.py", line 550, in get_conn
    kube_client = await self._load_config() or async_client.ApiClient()
                  ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/airflow/.local/lib/python3.11/site-packages/airflow/providers/cncf/kubernetes/hooks/kubernetes.py", line 497, in _load_config
    await async_config.load_kube_config(
  File "/usr/local/airflow/.local/lib/python3.11/site-packages/kubernetes_asyncio/config/kube_config.py", line 566, in load_kube_config
    loader = _get_kube_config_loader_for_yaml_file(
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/airflow/.local/lib/python3.11/site-packages/kubernetes_asyncio/config/kube_config.py", line 530, in _get_kube_config_loader_for_yaml_file
    return KubeConfigLoader(
           ^^^^^^^^^^^^^^^^^
  File "/usr/local/airflow/.local/lib/python3.11/site-packages/kubernetes_asyncio/config/kube_config.py", line 148, in __init__
    self.set_active_context(active_context)
  File "/usr/local/airflow/.local/lib/python3.11/site-packages/kubernetes_asyncio/config/kube_config.py", line 159, in set_active_context
    context_name = self._config['current-context']
                   ~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^
  File "/usr/local/airflow/.local/lib/python3.11/site-packages/kubernetes_asyncio/config/kube_config.py", line 415, in __getitem__

```

### What you think should happen instead

The config file must be regenerated every time the trigger wakes up. The content of the file should probably be saved in the operator to avoid fetching it from the cluster every time.

### How to reproduce

Submit an EksPodOperator with `deferrable=True`

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by Taragolis** [comment]
Did you check with [latest amazon provider](https://airflow.apache.org/docs/apache-airflow-providers-amazon/stable/index.html)?
`EksPodOperator` depend on [`apache-airflow-providers-cncf-kubernetes`](https://airflow.apache.org/docs/apache-airflow-providers-cncf-kubernetes/stable/index.html) so it worthwhile to update it.

**Comment by 0x26res** [comment]
> Did you check with [latest amazon provider](https://airflow.apache.org/docs/apache-airflow-providers-amazon/stable/index.html)? `EksPodOperator` depend on [`apache-airflow-providers-cncf-kubernetes`](https://airflow.apache.org/docs/apache-airflow-providers-cncf-kubernetes/stable/index.html) so it worthwhile to update it.

I'm not able to, MWAA dictates which version of the provider I can use.

**Comment by Taragolis** [comment]
[Amazon Managed Workflows for Apache Airflow: Installing Python dependencies](https://docs.aws.amazon.com/mwaa/latest/userguide/working-dags-dependencies.html)

**Comment by 0x26res** [comment]
Thanks for sharing the link @Taragolis 

As mentioned there, MWAA restrict which version can be used. In particular I'm using MWAA 2.7.2, which uses [this constraint file](https://raw.githubusercontent.com/apache/airflow/constraints-2.7.2/constraints-3.11.txt), and means I can only use `apache-airflow-providers-amazon==8.7.1`.


The only way to upgrade would be to go to version [v2.8.1](https://airflow.apache.org/docs/apache-airflow/2.8.1/release_notes#airflow-2-8-1-2024-01-19) with this [updated constraint file](https://raw.githubusercontent.com/apache/airflow/constraints-2.8.1/constraints-3.11.txt), and `apache-airflow-providers-amazon==8.16.0`.

I'll give it a try, but I doubt it will fix the issue as it's not mentioned as fixed in the release notes https://airflow.apache.org/docs/apache-airflow-providers-amazon/stable/changelog.html

**Comment by Taragolis** [comment]
It is also covered by [Amazon Managed Workflows for Apache Airflow: Specifying newer provider packages](https://docs.aws.amazon.com/mwaa/latest/userguide/connections-packages.html#connections-packages-newer-packages)

> [!IMPORTANT]
> https://github.com/apache/airflow/issues only intend to bugs which could be reproduced into the latest version of Airflow and latest versions of Airflow Providers. If you unable to update version of provider/airflow itself. You might need to contact MWAA support.

**Comment by 0x26res** [comment]
@Taragolis I tried to reproduce locally with the latest version of airflow (instead of using MWAA).

But this other [bug](https://github.com/apache/airflow/discussions/37178#discussioncomment-8823983) gets in the way of my testing 
```
pods is forbidden: User \"system:anonymous\" cannot list resource \"pods\" in API group
```

There is a work around [suggested here](https://github.com/apache/airflow/discussions/37178#discussioncomment-8823983) that consist in reducing logging. I've set my logging config accordingly in `airflow.cfg`
```
[logging]
logging_level = ERROR
```
But it doesn't work in the local environment. 

Which means I can't reproduce this issue, with the latest version of airflow, until the previous one is fixed.
